### PR TITLE
DOC: Update Zenodo info

### LINKED ISSUE #10611: Update Zenodo records?
It looks like the last Astropy version with a Zenodo record was v3.0.5 (https://zenodo.org/record/2556700#.XyF4wRNKjUJ). Do we want to upload records for the last few releases, and add "update Zenodo record" to the release procedures? Or are we ok with not updating Zenodo?

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request updates Zenodo release instructions and adds a Zenodo badge to README.

Fix #10611

### CODE REVIEW

File: docs/development/releasing.rst [hunk 1]
```diff
@@ -210,22 +210,15 @@ Post-Release procedures
...
-   in the form: the **Publication date** should be set to the date the tar
-   file was uploaded to PyPI, the **Title** should be updated to include the
-   new version number, and the **Version** should be updated to include the
-   version number (with no ``v`` prefix). Once you are happy with the changes,
-   click **Save**, then **Publish**.
+#. Upload the release to Zenodo by creating a GitHub Release off the GitHub tag.
+   Click on the tag in https://github.com/astropy/astropy/tags and then click on
+   the "Edit tag" button on the upper right. The release title is the same as the
+   tag. In the description, you can copy and paste a description from the previous
+   release, as it should be a one-liner that points to `CHANGES.rst`. When you
```
Comment (adrn) [inline docs/development/releasing.rst:217]: Docs failure is real - maybe:
```suggestion
   release, as it should be a one-liner that points to ``CHANGES.rst``. When you
```

**Review by adrn — APPROVED** [review]
Otherwise LGTM!

File: docs/development/releasing.rst (hunk 1 above)
Comment (pllim) [reply docs/development/releasing.rst]: Ops, you are right!

**Comment by adrn** [comment]
Thanks!

**Pull request merged**